api/types/container/hostconfig_windows.go

```diff
@@ -25,11 +27,11 @@ func (i Isolation) IsValid() bool {
 // NetworkName returns the name of the network stack.
 func (n NetworkMode) NetworkName() string {
 	if n.IsDefault() {
-		return "default"
+		return network.NetworkDefault
```

Comment:
Would it make sense to define the types in this package? (Given that `NetworkMode` is also defined here 🤔)

Reply:
I went back and forth and I couldn't find any prevailing argument. But here's my final reasoning on that:

- These consts are about predefined networks. As a client user, I'd expect networking stuff, including consts for predefined networks, to be defined in the ` network` package ;
- They're used in other places than just this package ;
- `NetworkMode` is about the main network _of a container_, these consts aren't related to containers ;

Reply:
@thaJeztah Are you still on the fence on that?

Reply:
> Are you still on the fence on that?

Yeah, probably shouldn't block the PR though.

It's just very messy because

- we conflated "mode" and "network-names" (which was intentional, but implementation of that wasn't great in hindsight)
- "default" is horrible, or more specifically, horrible because it's used in some places, but may not work in other places, which also relates to the above (does `docker network create --driver=default` work?)
- for places where we're currently checking for "names" ("modes"), we should more look at _properties_ of the network (is this network "attachable", does it support "x", can it be "created" / instantiated?)
- we still need to look at making some checks platform-independent; i.e. we should probably not allow creating a network named `bridge` on Windows, or vice-versa, a network named `transparent` (or `nat`) on Linux. We currently define some reserved names based on the platform the daemon is running on, but I think we should make those a general "reserved names" (or something like that)

> (does `docker network create --driver=default` work?)

Answering myself: it doesn't

```bash
docker network create --driver=default hellonetwork
Error response from daemon: plugin "default" not found
```